### ISSUE TITLE
EWS Throttling issues being rolled out worldwide on 365

### ISSUE THREAD

**Issue by vlw007** [OfficeDev/ews-managed-api#293]
Starting April 6, we are seeing EWS throttling issues when there is barely any EWS activity on a tenant.   

The error looks like this:
_[You have exceeded the available concurrent connections for your account.  Try again once your other requests have completed.]
ErrorDetails; Key=[Policy], Value=[MaxConcurrency]
ErrorDetails; Key=[MaxConcurrencyLimit], Value=[27]
ErrorDetails; Key=[ErrorMessage], Value=[This operation exceeds the throttling budget for policy part 'MaxConcurrency', policy value '27',  Budget type: 'Ews'.  Suggested backoff time 0 ms.]_

When this step executes, Our SaaS platform has only a single thread (no concurrency) and our backoff time is way larger than zero ms.

We made the request to remove throttling policies for 90 days.  The support request suggested that we wait 15 minutes for the policy to be updated

We waited two hours and there does not appear to be any difference.  The same error appears.  Another application using EWS has also started having problems at almost the exact same time.  

Does anyone notice this happening too?

**Comment by cmcveen** [comment]
Yes, I have noticed this same issue over the past couple of days in a few tenants.

**Comment by mariusan** [comment]
Same here on a couple of tenants

Regards
Marius

**Comment by joshmadams** [comment]
I am seeing this error in our tenant as well.

**Comment by PhilbyTheGreat** [comment]
We are seeing this in our tenant too.

**Comment by mariusan** [comment]
Below is the result of a trace. This is in response to the very first call and there are no other uses of the account.

```
<Trace Tag="EwsResponse" Tid="5" Time="2023-04-13 06:00:12Z" Version="[2.2.1.0](http://2.2.1.0/)">
  <?xml version="1.0" encoding="utf-8"?>
  <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
    <s:Body>
      <s:Fault>
        <faultcode xmlns:a="http://schemas.microsoft.com/exchange/services/2006/types">a:ErrorExceededConnectionCount</faultcode>
        <faultstring xml:lang="en-US">You have exceeded the available concurrent connections for your account.  Try again once your other requests have completed.</faultstring>
        <detail>
          <e:ResponseCode xmlns:e="http://schemas.microsoft.com/exchange/services/2006/errors">ErrorExceededConnectionCount</e:ResponseCode>
          <e:Message xmlns:e="http://schemas.microsoft.com/exchange/services/2006/errors">You have exceeded the available concurrent connections for your account.  Try again once your other requests have completed.</e:Message>
          <t:MessageXml xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
            <t:Value Name="Policy">MaxConcurrency</t:Value>
            <t:Value Name="MaxConcurrencyLimit">27</t:Value>
            <t:Value Name="ErrorMessage">This operation exceeds the throttling budget for policy part 'MaxConcurrency', policy value '27',  Budget type: 'Ews'.  Suggested backoff time 0 ms.</t:Value>
          </t:MessageXml>
        </detail>
      </s:Fault>
    </s:Body>
  </s:Envelope>
</Trace>
```

**Comment by halliballi123** [comment]
same here, problem started yesterday evening.

**Comment by joshmadams** [comment]
This issue started around 7am EST yesterday (4/12/23) for us.

**Comment by Yup-maker** [comment]
5 customers also have this problem with Public Folders. (However, I myself can not reproduce it)

This is my Exchange Online Version: 15.20.6277.038 V2018_01_08  
This is one of my customers: 15.20.6298.028 V2018_01_08

**Comment by halliballi123** [comment]
> 5 customers also have this problem with Public Folders. (However, I myself can not reproduce it)
> 
> This is my Exchange Online Version: 15.20.6277.038 V2018_01_08 This is one of my customers: 15.20.6298.028 V2018_01_08

In our case it's also public folders. We create contacts in public folders.
Version is 15.20.6298.30

**Comment by mariusan** [comment]
Might be the same issue:
https://admin.microsoft.com/Adminportal/Home#/servicehealth/:/alerts/EX540990

**Comment by vlw007** [comment]
Are you able to get the Exchange Online Version via an API call ?

________________________________

**Comment by Yup-maker** [comment]
Yes by using ExchangeServerInfo: service.ServerInfo?.ToString()

**Comment by vlw007** [comment]
Are you able to get the Exchange Online Version via an API call ?

________________________________

**Comment by ToS0** [comment]
Same here - since yesterday

**Comment by viebrix** [comment]
We also had the issue with public folder, but it seems it is right now solved

**Comment by Andy1860** [comment]
Same here, seems to be solved.

**Comment by viebrix** [comment]
Greg Taylor from the Exchange Teams asked here why and how people use public folders:
[techcommunity - exchange-team-blog](https://techcommunity.microsoft.com/t5/exchange-team-blog/upcoming-api-deprecations-in-exchange-web-services-for-exchange/bc-p/3797269#M36172)
I think this is a great chance to help the team to improve exchange for us.

**Comment by halliballi123** [comment]
problem still there

**Comment by viebrix** [comment]
@halliballi123 
I'm only user and not experience in this topic. But did you try to request to remove throttling policies for 90 days, as vlw007 is describing in his first post? We also did this step last week. Maybe this makes a difference?

**Comment by halliballi123** [comment]
I don't know how and where to put that request

**Comment by viebrix** [comment]
Found this description in the web:
    Sign in to Microsoft 365 Admin Center and click on the Help icon in the top corner of the screen.
    Search for EWS and select the Increase EWS Throttling Policy article that is suggested.In the Run diagnostics section that is displayed, select Run Tests.   If the diagnostic returns a The Exchange Web Services are throttled message, you have the option to change (ease) the policy to allow your migration to complete.
        Choose the Duration (Days) that you want to relax the policy for from the drop down menu. You can choose between 20, 60 and 90 days. Place a tick in the checkbox to acknowledge that want to make the specified changes.
        Select Update Settings to apply the changes to the policy. Changes can take up to 30 minutes to take effect.
(source of text, support.cloudm.io)

**Comment by Andy1860** [comment]
> Found this description in the web: Sign in to Microsoft 365 Admin Center and click on the Help icon in the top corner of the screen. Search for EWS and select the Increase EWS Throttling Policy article that is suggested.In the Run diagnostics section that is displayed, select Run Tests. If the diagnostic returns a The Exchange Web Services are throttled message, you have the option to change (ease) the policy to allow your migration to complete. Choose the Duration (Days) that you want to relax the policy for from the drop down menu. You can choose between 20, 60 and 90 days. Place a tick in the checkbox to acknowledge that want to make the specified changes. Select Update Settings to apply the changes to the policy. Changes can take up to 30 minutes to take effect. (source of text, support.cloudm.io)

I already tried this last week when the problems started with no improvement.

**Comment by Andy1860** [comment]
I think you just have to wait until it works again. We haven't had any more problems since yesterday afternoon.

Last update in M365 Health Center:

April 17, 2023 7:24 PM
Title: Some users may be unable to access, view, add, or select public folders and calendars in Outlook on the web
User impact: Users may be unable to access, view, add, or select public folders and calendars in Outlook on the web.
**Current status: We've determined the status of the fix deployment to currently stand at 80 percent saturation, and we’ll provide an estimated timeline for resolution once available.**
Scope of impact: Some users attempting to perform multiple public folder or calendar related actions in Outlook on the web may be impacted.
Start time: Tuesday, April 11, 2023, 1:37 PM (11:37 AM UTC)
Root cause: A recent service update contained a code issue that’s preventing users from accessing, viewing, adding, or selecting public folders and calendars in Outlook on the web.
Next update by: Tuesday, April 18, 2023, 8:00 PM (6:00 PM UTC)

**Comment by halliballi123** [comment]
> I think you just have to wait until it works again. We haven't had any more problems since yesterday afternoon.
> 
> Last update in M365 Health Center:
> 
> April 17, 2023 7:24 PM Title: Some users may be unable to access, view, add, or select public folders and calendars in Outlook on the web User impact: Users may be unable to access, view, add, or select public folders and calendars in Outlook on the web. **Current status: We've determined the status of the fix deployment to currently stand at 80 percent saturation, and we’ll provide an estimated timeline for resolution once available.** Scope of impact: Some users attempting to perform multiple public folder or calendar related actions in Outlook on the web may be impacted. Start time: Tuesday, April 11, 2023, 1:37 PM (11:37 AM UTC) Root cause: A recent service update contained a code issue that’s preventing users from accessing, viewing, adding, or selecting public folders and calendars in Outlook on the web. Next update by: Tuesday, April 18, 2023, 8:00 PM (6:00 PM UTC)

thanks, we tried the fix but still no improvement, so maybe you are right. We have to wait ...

**Comment by halliballi123** [comment]
finally it is working again ...
